doc/guide.md

```diff
@@ -277,7 +277,87 @@ returncode_t libtock_sensor_read(libtock_sensor_callback_reading cb) {
 
 ## Synchronous APIs
 
-Most system call interfaces will want to provide a synchronous API as well.
+Most system call interfaces will want to provide a synchronous API as well. This
+requires another file for syscalls.
+
+All synchronous APIs MUST use the
+[Yield-WaitFor](https://book.tockos.org/trd/trd104-syscalls.html#413-yield-waitfor)
+(`yield_wait_for()`) variant of the yield syscall. This ensures predictable
+behavior for `libtock-sync` users because Yield-WaitFor ensures that no other
+application upcall will run until the synchronous API has finished.
```

Comment:
Do we intend to move all the other synchronous APIs to Yield-WaitFor in this same PR? Otherwise the docs and the code will be out-of-sync.

Reply:
I'm not really excited about another massive change that requires touching every driver. I think we need to make forward progress on YWF and we haven't been able to so far. I would like to open a tracking issue for all of the drivers and convert them over time.

Reply:
That's totally reasonable. Maybe the doc should say that this is the ideal and synchronous drivers will be changed over to this over time?